instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -197,25 +195,22 @@ def _patched_api_call(self, original_func, instance, args, kwargs):
             _safe_invoke(extension.before_service_call, span)
             self._call_request_hook(span, call_context)
 
-            token = context_api.attach(
-                context_api.set_value(_SUPPRESS_HTTP_INSTRUMENTATION_KEY, True)
-            )
-
-            result = None
             try:
-                result = original_func(*args, **kwargs)
-            except ClientError as error:
-                result = getattr(error, "response", None)
-                _apply_response_attributes(span, result)
-                _safe_invoke(extension.on_error, span, error)
-                raise
-            else:
-                _apply_response_attributes(span, result)
-                _safe_invoke(extension.on_success, span, result)
+                with suppress_http_instrumentation():
+                    result = None
```

Comment:
I tried to preserve the logic with the exception catching and finally here but a second set of eyes would be good, especially given that I don't know if calling `_safe_invoke(extension.after_service_call)` after de-suppressing instrumentation is on purpose or it just happened to be like that

Reply:
This looks fine